src/QoiSharp/QoiEncoder.cs

```diff
@@ -15,6 +15,12 @@ public static class QoiEncoder
     /// <returns>Encoded image.</returns>
     /// <exception cref="QoiEncodingException">Thrown when image information is invalid.</exception>
     public static byte[] Encode(QoiImage image)
+    {
+        var bytes = new byte[QoiCodec.HeaderSize + QoiCodec.Padding.Length + (image.Width * image.Height * (byte)image.Channels)];
+        return bytes[..Encode(image, bytes)];
+    }
+
+    public static int Encode(QoiImage image, Span<byte> bytes)
```

Comment:
API compatible change, allows users to pre-allocate buffers which (should?) provide a zero-allocation way of encoding multiple images. Just allocate a single (large enough) buffer up front.

You can also encode multiple images to the same byte[] as long as you `Slice` it appropriately.

Reply:
An alternative to this approach is to mimic the API here: https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm.trycomputehash?view=net-6.0